gradle/dependencies.gradle

```diff
@@ -3,8 +3,8 @@ ext {
 
   androidVersions = [
       minSdkVersion           : 21,
-      targetSdkVersion        : 30,
-      compileSdkVersion       : 30
+      targetSdkVersion        : 31,
+      compileSdkVersion       : 31
```

Comment:
This will also force consuming apps to target `31` which I don't think we want to do just yet.

Reply:
What if we only upgrade the qa/test apps to use 31 to verify the compatibility but leave the actual SDK to still target `30`?

Reply:
> This will also force consuming apps to target `31` which I don't think we want to do just yet.

Are you sure it works that way?

Reply:
I might be wrong. But `lifecycle-runtime-ktx` does require compile version of 31. This is what happens if you'd try to use 30 instead:
```
The minCompileSdk (31) specified in a
dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
is greater than this module's compileSdkVersion (android-30).
Dependency: androidx.lifecycle:lifecycle-runtime-ktx:2.4.0.
AAR metadata file: /home/lukas/.gradle/caches/transforms-2/files-2.1/d48d5789dbc4707860dc22d65d156899/jetified-lifecycle-runtime-ktx-2.4.0/META-INF/com/android/build/gradle/aar-metadata.properties.
```

Reply:
Looks like this flag is set in [`AarMetadata`](https://developer.android.com/reference/tools/gradle-api/7.1/com/android/build/api/variant/AarMetadata). I was under an impression that this is driven by the compile version but that's not the case, sorry!

Reply:
But we'll still need to revert this dependency bump, we don't want to force `31` with our transitive dependencies.

Reply:
This affects `compileSdkVersion` only, which should be safe to bump. E.g. 1TAP uses the latest lifecycle library and thus is required to specify `compileSdkVersion` 31, but `targetSdkVersion` is still 30.

Reply:
Thanks @Zayankovsky, so with this update we would enforce compile version on developers, not target version. This is unfortunate but we might not have to consider this a blocker.

Let's make sure that we specify in the changelog entry that this requires `31` as `compileSdkVersion` due to the `androidx.lifecycle:lifecycle-runtime-ktx:2.4.0` bump.

cc @Guardiola31337